tests/rados/test_mon_system_operations.py

```diff
@@ -0,0 +1,268 @@
+        raise Exception("mon service not added error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    # Checking if the new mon added is part of Mon Quorum
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if alt_mon_host.hostname not in quorum:
+        log.error(
+            f"selected host : {alt_mon_host.hostname} does not have mon as part of Quorum"
+        )
+        raise Exception("Mon not in quorum error")
+
+    log.info("Completed workflow 1. Addition of new mon daemon")
+
+    log.info("---------- starting workflow 2 - Removal of new mons --------")
+    # Setting the mon service as unmanaged
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    # Removing mon service
+    if not mon_obj.remove_mon_service(host=alt_mon_host.hostname):
+        log.error("Could not remove the new mon added previously")
+        raise Exception("mon service not removed error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if alt_mon_host.hostname in quorum:
+        log.error(
+            f"selected host : {alt_mon_host.hostname} is present as part of Quorum"
+        )
+        raise Exception("Mon in quorum error post removal")
+
+    log.info("Completed Scenario 2")
+
+    log.info(
+        "---------- starting workflow 3 - Removal of mons during OSD down --------"
+    )
+
+    # fetching all the OSDs on the cluster and selecting 1 OSD at random to be shut down.
+    out, _ = cephadm.shell(args=["ceph osd ls"])
+    osd_list = out.strip().split("\n")
+
+    osd_id = int(random.choice(osd_list))
+    if not rados_obj.change_osd_state(action="stop", target=osd_id):
+        log.error(f"Could not stop OSD : {osd_id}")
+        raise Exception("OSD not stopped error")
+
+    # Selecting one mon host to be removed at random during OSD down
+    test_mon_host = random.choice(init_mon_nodes)
+
+    # Setting the mon service as unmanaged
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    # Removing mon service
+    if not mon_obj.remove_mon_service(host=test_mon_host.hostname):
+        log.error("Could not remove the new mon added previously")
+        raise Exception("mon service not removed error")
+
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if test_mon_host.hostname in quorum:
```

Comment:
The condition here needs modification as it is removal of mon scenario.

Reply:
Corrected the logging wordings. The conditional checks are still Correct. it was the log lines that needed updates.